### PR TITLE
Ap 5040 add care of question

### PR DESCRIPTION
**Pull request by kmahern**
## What

[AP-5040](https://dsdmoj.atlassian.net/browse/AP-5040)

Add 'Do you want to add a ‘care of’ recipient for your client’s mail?' question

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

[AP-5040]: https://dsdmoj.atlassian.net/browse/AP-5040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/providers/correspondence_address/care_ofs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+      def update
+        @form = Addresses::CareOfForm.new(form_params)
+
+        render :show unless save_continue_or_draft(@form)
+      end
+
+    private
+
+      def legal_aid_application
```
Comment (colinbruce) [inline app/controllers/providers/correspondence_address/care_ofs_controller.rb:18]: I'm not sure this is needed, a legal_aid_application variable in one of the lover level controllers, if you need it to get the applicant you could replace it with 

```ruby
    def applicant
      @applicant ||= @legal_aid_application.applicant
    end
```
or, if you only need the applicant for the address?

```ruby
    def applicant
      @address ||= @legal_aid_application.applicant.address
    end
```

File: spec/forms/addresses/care_of_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    end
+
+    context "with missing organisation name" do
+      let(:care_of) { "organisation" }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).to be_valid
```
Comment (colinbruce) [inline spec/forms/addresses/care_of_form_spec.rb:202]: The test description doesn't match the expectation invalid or valid

File: spec/forms/addresses/care_of_form_spec.rb [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    end
+
+    context "with missing organisation name" do
+      let(:care_of) { "organisation" }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).not_to be_valid
```
Comment (colinbruce) [inline spec/forms/addresses/care_of_form_spec.rb:105]: The test description doesn't match the expectation invalid or valid

File: spec/forms/addresses/care_of_form_spec.rb [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    end
+
+    context "with missing person name" do
+      let(:care_of) { "person" }
+      let(:care_of_last_name) { "Smith" }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).not_to be_valid
```
Comment (colinbruce) [inline spec/forms/addresses/care_of_form_spec.rb:89]: The test description doesn't match the expectation invalid or valid

File: spec/forms/addresses/care_of_form_spec.rb [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    end
+
+    context "with no answer chosen" do
+      let(:care_of) { nil }
+      let(:care_of_last_name) { nil }
+      let(:care_of_first_name) { nil }
+      let(:care_of_organisation_name) { nil }
+
+      it "is invalid" do
+        expect(instance).not_to be_valid
```
Comment (colinbruce) [inline spec/forms/addresses/care_of_form_spec.rb:73]: The test description doesn't match the expectation invalid or valid

**Review by colinbruce — COMMENTED** [review]
Looking a good, a few test descriptions need updating and an observation about the controller methods

File: app/controllers/providers/correspondence_address/care_ofs_controller.rb (hunk 1 above)
Comment (kmahern) [reply app/controllers/providers/correspondence_address/care_ofs_controller.rb]: 👍 updated

File: spec/forms/addresses/care_of_form_spec.rb (hunk 3 above)
Comment (kmahern) [reply spec/forms/addresses/care_of_form_spec.rb:105]: I think this description does match(is invalid not_to be_valid), or is it not clear?

File: spec/forms/addresses/care_of_form_spec.rb (hunk 4 above)
Comment (kmahern) [reply spec/forms/addresses/care_of_form_spec.rb:89]: I think this description does match(is invalid not_to be_valid), or is it not clear?

File: spec/forms/addresses/care_of_form_spec.rb (hunk 5 above)
Comment (kmahern) [reply spec/forms/addresses/care_of_form_spec.rb:73]: I think this description does match(is invalid not_to be_valid), or is it not clear?

**Review by colinbruce — APPROVED** [review]
LGTM :+1:
Sorry about the heavy handed copy pasting! 🤦‍♂️

**Pull request merged**